build/crossorigin-auto-attribute-plugin.js

```diff
@@ -29,7 +29,7 @@ class CrossoriginAutoAttributePlugin {
         source => {
           return source.replace(
             /script\.src = [^;]*;/,
-            '$& if (!script.src.match(/^https:\\/\\//)) {script.crossOrigin = void 0;}'
+            '$& if (!script.src.indexOf(window.location.origin)) {script.crossOrigin = void 0;}'
```

Comment:
`indexOf()` will return a 0 if it starts at the beginning, so should we compare with -1 instead?

Reply:
That's actually how it's intended to work.

When expanded, the logic looks something like this:

```js
var isSameOrigin = script.src.startsWith(window.location.origin);
// equivalent to:
var isSameOrigin = script.src.indexOf(window.location.origin) === 0;
// equivalent to:
var isSameOrigin = !(script.src.indexOf(window.location.origin) !== 0);
// equivalent to:
var isSameOrigin = !(Boolean(script.src.indexOf(window.location.origin)));
// equivalent to:
var isSameOrigin = !(script.src.indexOf(window.location.origin));

if (isSameOrigin) {
  removeCrossOriginAttribute();
}

```

I picked the smallest implementation (excluding `startsWith` because that would require polyfills) since it's going directly into the webpack bootstrap code and assigning variables, etc. is probably a bad idea. Also I don't think Uglify will be able to perform the simplification above.

Reply:
Maybe add a comment then?

Reply:
There's already a fairly extensive comment above that describes what this plugin does and why it is necessary. With that context, I don't think there should be any confusion.